src/traces/sankey/attributes.js

```diff
@@ -0,0 +1,168 @@
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the vertical domain of this `sankey` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        }
+    },
+
+    orientation: {
+        valType: 'enumerated',
+        values: ['v', 'h'],
+        dflt: 'h',
+        role: 'style',
+        description: 'Sets the orientation of the Sankey diagram.'
+    },
+
+    nodepad: {
+        valType: 'number',
+        min: 0,
+        dflt: 20,
+        role: 'style',
+        description: 'Sets the padding (in px) between the `nodes`.'
+    },
+
+    nodethickness: {
+        valType: 'number',
+        min: 1,
+        dflt: 20,
+        role: 'style',
+        description: 'Sets the thickness (in px) of the `nodes`.'
+    },
+
+    valueformat: {
+        valType: 'string',
+        dflt: '.3s',
+        role: 'style',
+        description: [
+            'Sets the value formatting rule using d3 formatting mini-language',
+            'which is similar to those of Python. See',
+            'https://github.com/d3/d3-format/blob/master/README.md#locale_format'
+        ].join(' ')
+    },
+
+    valuesuffix: {
+        valType: 'string',
+        dflt: '',
+        role: 'style',
+        description: [
+            'Adds a unit to follow the value in the hover tooltip. Add a space if a separation',
+            'is necessary from the value.'
+        ].join(' ')
+    },
+
+    arrangement: {
+        valType: 'enumerated',
+        values: ['snap', 'perpendicular', 'freeform', 'fixed'],
+        dflt: 'snap',
+        role: 'style',
+        description: [
+            'If value is `snap` (the default), the node arrangement is assisted by automatic snapping of elements to',
+            'preserve space between nodes specified via `nodepad`.',
+            'If value is `perpendicular`, the nodes can only move along a line perpendicular to the flow.',
+            'If value is `freeform`, the nodes can freely move on the plane.',
+            'If value is `fixed`, the nodes are stationary.'
+        ].join(' ')
+    },
+
+    textfont: fontAttrs,
+
+    nodes: {
+        _isLinkedToArray: 'node',
+        label: {
+            valType: 'string',
+            dflt: '',
+            role: 'info',
+            description: 'The shown name of the node.'
+        },
+        parallel: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 1,
+            description: 'Node position parallel to the flow, e.g. left/right if `orientation` is `horizontal`.'
+        },
+        perpendicular: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            max: 1,
+            description: 'Node position perpendicular to the flow, e.g. up/down if `orientation` is `horizontal`.'
```

Comment:
Do you want to say something here about the automatic layout system - how these get chosen if you don't specify them? Doesn't have to be detailed at all, just enough to let people know that you don't *have* to put a value in at all.